test/others/criu-ns/run.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+
+import multiprocessing
+import os
+import pty
+import shutil
+import subprocess
+import sys
+import time
+
+
+CRIU_BIN = "../../../criu/criu"
+CRIU_NS = "../../../scripts/criu-ns"
+
+
+def check_dumpdir():
+    if os.path.isdir("dumpdir"):
+        shutil.rmtree("dumpdir")
+    os.mkdir("dumpdir", 0o755)
+
+
+def create_pty():
+    fd_m, fd_s = pty.openpty()
+    return (os.fdopen(fd_m, "wb"), os.fdopen(fd_s, "wb"))
+
+
+def test_dump_and_restore_with_shell_job():
+    check_dumpdir()
+    os.setsid()
+
+    with open("running", "w") as file:
+        pass
+    pid = os.fork()
+    if pid == 0:
+        while True:
+            if not os.access("running", os.F_OK):
+                sys.exit(0)
+            time.sleep(1)
+
+    cmd = [CRIU_NS, "dump", "-D", "dumpdir", "-v", "--shell-job",
+           "-t", str(pid), "--criu-binary", CRIU_BIN]
+    ret = subprocess.Popen(cmd).wait()
```

Comment:
The code that used to run `criu` is appears to be duplicated several times. It might be good to move this code in a separate function.

Reply:
Yeah, I agree. But, the `pid` is retrieved on the fly (through `os.fork` and `multiprocessing.Process.start`. If I moved this code block into a separate function that take `pid` as an argument, it kind of feel like I'm disturbing the flow of execution or readability :)

Reply:
@warusadura Reducing code duplication is about improving maintainability (e.g., changing `-v` to `-v4` everywhere could be done by changing a single line).

Reply:
understood, thanks!

Reply:
remove the code duplication of restore processes as well.